### PR TITLE
Split Windows CI into core and non-core steps

### PR DESCRIPTION
**Pull request by rnorth**

### CODE REVIEW

**Review by bsideup — APPROVED** [review]

**Review by kiview — APPROVED** [review]
Since we have a high overlap between the stages, would it make sense to extract them into a [template](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#insert-a-template)?

Maybe we can think about it, if we would get a fourth task with duplicate elements.

**Comment by bsideup** [comment]
@kiview I would rather spend time on GHA hosted workers migration then :)

File: azure-pipelines.yml [hunk 1]
```diff
@@ -18,17 +18,32 @@ jobs:
...
+    condition: eq(variables['Agent.OS'], 'Windows_NT')
+    displayName: Build & test (Windows - all non-core modules)
+    env:
+      AWS_ACCESS_KEY_ID: $(aws.accessKeyId)
+      AWS_SECRET_ACCESS_KEY: $(aws.secretAccessKey)
+    inputs:
+        gradleWrapperFile: 'gradlew'
+        jdkVersionOption: '1.11'
+        options: '--no-daemon --continue'
+        tasks: 'clean check -x testcontainers:test'
```
Comment (gesellix) [inline azure-pipelines.yml:47]: Is `testcontainers:test` intentional? I would've expected `testcontainers:check`.

**Pull request merged**